### PR TITLE
feat: add complete markdown preview

### PR DESCRIPTION
**Pull request by nikhilkalburgi**
**Description**

I have completed building the markdown preview.

https://github.com/asyncapi/vs-asyncapi-preview/assets/70331875/1ef192bd-4945-476f-87f8-c540e6fe62d6

 The following are the topics that should be taken care of in the future.
Order | Topics
-- | --
1 | Testing and Debugging the developed   Markdown Preview
2 | Enhancement in the Markdown Preview
3 | Building the Flowchart Preview
4 | Planning for the Class Diagram Preview
5 | Adding Support for export
6 | Adding Support for Avro Syntax
7 | Other functions like sync, etc.

I want to continue these tasks with the **GSOC24**. I will submit the proposal on the GSOC portal for this Project Idea as well as for the **AutoFix** feature. 🙏🏼Please consider my application, if these ideas are selected for GSOC. So that I can continue here with my full potential. 😊However, I can continue with the planning and design of this feature + research on the AutoFix feature (which I have already started).

Thank You for this Opportunity,
**Nikhil S Kalburgi**

### CODE REVIEW

File: src/PreviewMarkdown.ts [hunk 1]
```diff
@@ -93,25 +93,181 @@ function getWebviewContent(context: vscode.ExtensionContext, webview: vscode.Web
...
-      }
-      body {
-        color: #121212;
-        background-color: #fff;
-        word-wrap: break-word;
-      }
-      h1 {
-        color: #121212;
-      }
+        html{
```
Comment (ivangsa) [inline src/PreviewMarkdown.ts:96]: Consider moving this css to a separate file

File: src/components/Asyncapi.ejs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<%- include(path.infoPath,{...info,tagsPath: path.tagsPath}) %>
```
Comment (ivangsa) [inline src/components/Asyncapi.ejs:1]: Make sure this renders even with an incomplete/invalid spec

**Review by ivangsa — APPROVED** [review]
Good job. Consider this canges for next version. I'm merging this.. Thanks!!

**Comment by nikhilkalburgi** [comment]
HI @ivangsa,

I will take care of the changes you have suggested along with the development of the flowchart and class diagram. I have also submitted the proposal to GSOC 24 Asyncapi Organization. Can you please take some time to review it and share some feedback on that?

**Comment by ivangsa** [comment]
Hi @nikhilkalburgi 

you have done a great job on this feature, not finished yet but good work..

It's probably not me how is going to mentor the gsoc as a new team is going to get in charge of this project...

Rest assured that your proposal would be evaluated acordingly... as I'm in contact with all the team..

By the time being, if you enjoy it.. keep contributing to opensource..

Should you need anything else I'll be arround here in github

Happy coding to you!!

**Comment by nikhilkalburgi** [comment]
Hi @ivangsa,

I hope you are doing great. This is to inform you that I am now selected as GSOC 2024 contributor for this project idea. My mentor is Ace (According to the GSOC Dashboard). 

Should I continue with work that we planned together or should I follow up with my mentor in GSOC with a different working approach?

Thank you for the recognition.

**Comment by ivangsa** [comment]
hi @nikhilkalburgi, congratulations!!

you should follow Ace instructions for anything GSOC related... and you can continue this line of work
I will continue to review your PRs

I'm in direct contact with @AceTheCreator in the slack channel, so no worries about it..

This being said, for me you are just another fellow opensource developer and contributor 

so Happy Coding!!

**Comment by nikhilkalburgi** [comment]
Thank you so much 🙂 . I will continue my work here.

**Pull request merged**